presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -224,12 +224,12 @@ public ConnectorTableLayoutResult getTableLayoutForConstraint(
                 new IcebergTableLayoutHandle.Builder()
                         .setPartitionColumns(ImmutableList.copyOf(getPartitionKeyColumnHandles(icebergTable, typeManager)))
                         .setDataColumns(toHiveColumns(icebergTable.schema().columns()))
-                        .setDomainPredicate(constraint.getSummary().transform(IcebergAbstractMetadata::toSubfield))
+                        .setDomainPredicate(constraint.getSummary().simplify().transform(IcebergAbstractMetadata::toSubfield))
```

Comment:
We should be careful about the `simplify()` here, for Iceberg could not support a large value list for IN predicate. So the `unenforcedConstraint` in the returned `ConnectorTableLayoutResult` should be `constraint.getSummary()`.